components/script/dom/readablebytestreamcontroller.rs

```diff
@@ -2,89 +2,2001 @@
+                    cx,
+                    &self.global(),
+                    error.handle_mut(),
+                );
+                // Perform ! ReadableByteStreamControllerError(controller, e).
+                self.error(error.handle(), can_gc);
+
+                // Perform readIntoRequest’s error steps, given e.
+                read_into_request.error_steps(error.handle(), can_gc);
+
+                // Return.
+                return;
+            }
+        }
+
+        // Append pullIntoDescriptor to controller.[[pendingPullIntos]].
+        {
+            self.pending_pull_intos
+                .borrow_mut()
+                .push(pull_into_descriptor);
+        }
+
+        // Perform ! ReadableStreamAddReadIntoRequest(stream, readIntoRequest).
+        stream.add_read_into_request(read_into_request);
+
+        // Perform ! ReadableByteStreamControllerCallPullIfNeeded(controller).
+        self.call_pull_if_needed(can_gc);
     }
 
     /// <https://streams.spec.whatwg.org/#readable-byte-stream-controller-respond>
-    pub(crate) fn respond(&self, _bytes_written: u64) -> Fallible<()> {
-        todo!()
+    #[allow(unsafe_code)]
+    pub(crate) fn respond(&self, bytes_written: u64, can_gc: CanGc) -> Fallible<()> {
+        let cx = GlobalScope::get_cx();
+        {
+            // Assert: controller.[[pendingPullIntos]] is not empty.
+            let mut pending_pull_intos = self.pending_pull_intos.borrow_mut();
+            assert!(!pending_pull_intos.is_empty());
+
+            // Let firstDescriptor be controller.[[pendingPullIntos]][0].
+            let first_descriptor = pending_pull_intos.first_mut().unwrap();
+
+            // Let state be controller.[[stream]].[[state]].
+            let stream = self.stream.get().unwrap();
+
+            // If state is "closed",
+            // If bytesWritten is not 0, throw a TypeError exception.
+            if stream.is_closed() && bytes_written != 0 {
+                return Err(Error::Type("bytesWritten is not 0".to_owned()));
+            }
+
+            // Assert: state is "readable".
+            assert!(stream.is_readable());
+
+            // If bytesWritten is 0, throw a TypeError exception.
+            if bytes_written == 0 {
+                return Err(Error::Type("bytesWritten is 0".to_owned()));
+            }
+
+            // If firstDescriptor’s bytes filled + bytesWritten > firstDescriptor’s byte length,
+            // throw a RangeError exception.
+            if first_descriptor.bytes_filled.get() + bytes_written > first_descriptor.byte_length {
+                return Err(Error::Range(
+                    "bytes filled + bytesWritten > byte length".to_owned(),
+                ));
+            }
+
+            // Set firstDescriptor’s buffer to ! TransferArrayBuffer(firstDescriptor’s buffer).
+            first_descriptor.buffer = first_descriptor
+                .buffer
+                .transfer_array_buffer(cx)
```

Comment:
So you could change this to `transfer_array_buffer(cx, rval.handle_mut())` to remove the unsafe code.